File: java-files-corrected/10.java
class Identifiable {
    public String getId() {
        return name;
    }

    public String getPopulation() {
        return population;
    }
}

class City extends Identifiable {
    @Override
    public String getId() {
        return "City ID: " + name;
    }

    @Override
    public void describe() {
        System.out.println("City: " + name + ", Population: " + population);
    }
}

File: cpp-test-files/10.cpp
//Öröklődés2
#include <iostream>
#include <string>
using namespace std;

class Identifiable {
public:
    virtual string getId() const = 0;
    virtual ~Identifiable() {}
};

class City : public Identifiable {
protected:
    string name;
    int population;

public:
    City(const string& name, int population)
        : name(name), population(population) {}

    string getId() const override {
        return "City ID: " + name;
    }
};